.github/workflows/run-tests.yaml

```diff
@@ -0,0 +1,95 @@
+# This workflow will run all tests as well as pre-commit 
+
+name: Tests and linting
+on:
+  [push, pull_request]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    # All of these steps are just setup
+    - uses: actions/checkout@v2
+    - name: Setting MINICONDA_PATH
+      run: echo "MINICONDA_PATH=$HOME/miniconda" >> $GITHUB_ENV
+    - name: Setting OPENWPM_CONDA_PATH
+      run: echo "OPENWPM_CONDA_PATH=$MINICONDA_PATH/envs/openwpm" >> $GITHUB_ENV
+    # If the environment.yaml hasn't changed we just reuse the entire conda install
+    - id: cache
+      uses: actions/cache@v2
+      env:
+        cache-name: conda-cache
+      with:
+        path: ${{ env.MINICONDA_PATH }}
+        key: ${{ env.cache-name }}-${{ hashFiles('environment.yaml') }}
+
+    - name: Install conda
+      if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+      run: $GITHUB_WORKSPACE/scripts/install-miniconda.sh
+
+    - run: echo "$MINICONDA_PATH/bin" >> $GITHUB_PATH
+
+    - name: Install.sh (cache miss)
+      if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+      run: |
+        $GITHUB_WORKSPACE/install.sh
+    - name: Install.sh (cache hit)
+      if: ${{ steps.cache.outputs.cache-hit == 'true' }}
+      run: |
+        conda info --envs
```

Comment:
The `|` is not necessary for single-line commands, right?

Reply:
Yes, I have removed it :)